File: go/sublist/sublist.go
package sublist

// Relation is the relation of lists
type Relation string

// Sublist returns
func Sublist(a, b []int) Relation {
	if len(a) == 0 {
		if len(b) == 0 {
			return "equal"
		}
		return "sublist"
	}
	if len(b) == 0 {
		return "superlist"
	}
	if IsSublist(a, b) {
		if len(a) == len(b) {
			return "equal"
		}
		return "sublist"
	}
	if IsSublist(b, a) {
		return "superlist"
	}
	return "unequal"
}

// IsSublist returns true if a is a sublist of b
func IsSublist(a, b []int) bool {
	// find potential start of a in b
	for i := 0; i < len(b); i++ {
		if b[i] == a[0] {
			// found potential start, check rest of a
			for j := 1; j < len(a) && j+i < len(b); j++ {
				if a[j] != b[j+i] {
					break
				}
				if j == len(a)-1 {
					return true
				}
			}
		}
	}
	return false
}

File: python/sublist/sublist.py
EQUAL = 1
SUBLIST = 2
SUPERLIST = 3
UNEQUAL = 4

def check_lists(first_list, second_list):
  if len(first_list) == 0:
    if len(second_list) == 0:
      return EQUAL
    return SUBLIST
  if len(second_list) == 0:
    return SUPERLIST

  if first_list == second_list:
    return EQUAL

  if is_sublist(first_list, second_list):
    return SUBLIST
  
  if is_sublist(second_list, first_list):
    return SUPERLIST

  return UNEQUAL

def is_sublist(first_list, second_list):
  for i, num in enumerate(second_list):
    if num == first_list[0]:
      # found potential match, continue checking
      for j in range(1, len(first_list)):
        if j+i == len(second_list) or first_list[j] != second_list[j+i]:
          break
        if j == len(first_list) - 1:
          return True
  return False

    